.github/workflows/schedule.yml

```diff
@@ -113,7 +113,7 @@ jobs:
 
       - uses: actions/setup-python@v4
         with:
-          python-version: '3.8'
+          python-version: '3.10'
```

Comment:
I believe it's ok to change the version from 3.8, as its EOLed in 1 month, but why it's updated to 3.10? It makes sense to test on 3.9, as the tests include SDK, and 3.9 is still living.

Reply:
IMO, we shouldn't change the version in CI until we've actually dropped support for the old version(s) in the SDK.

Reply:
We use python 3.10 in a docker container, require 3.10 or higher in the development guide, and require 3.10 (the same as in Dockerfile) when generating server requirements. So, why should we use a different version when testing?
Additionally, I believe we should follow the same style when developing the product and when writing tests, but that can require using compatible python versions (e.g. using only `Union[X, Y]` or `X | Y`)

Reply:
Probably, it's better to do in a separate PR. Align versions on CI, think about what we're actually need to use for testing.

> python 3.10 in a docker container, require 3.10 or higher in the development guide, and require 3.10 (the same as in Dockerfile) when generating server requirements. So, why should we use a different version when testing?

This is the server part, which is supposed to be fixed and launched in a controlled environment (a docker container). This is not the same as the environment for tests and for SDK, which is, basically, can be arbitrary, but we limit the support to what we can reasonably maintain. I believe, we should maintain up to the oldest living version of python.